### DISCUSSION TITLE
Best practice for project structuring

### DISCUSSION THREAD

**Discussion by bigianb** [google/filament#4298] [Q&A]
I'm looking at using filament in a small program which I want to run on windows, mac, linux in the first instance. Generally I use cmake and I'm wondering what is the best way to include filament as a dependency.
Is there any support for findPackage in cmake? Is it best to just include it as a git submodule? Is it best to download the binaries and point cmake there ... if so what's the best way to point cmake to it (find package or just add the libs and include dirs manually?).

Are there any examples in github of projects using filament as a library.
I'm sure I can figure it out myself - but if there is a best practice I would rather follow that.

Ian

**Comment by jnmaloney** [comment]
Hi I'll just add some random notes about my experience building with filament so far.
- Filament is divided up into lots of libs and optional extensions, you need to decide which libraries you want to link to in your project
- Linux builds can be tricky, have to use `-fno-builtins` flag and build from source on Ubuntu 20 
I'm on Ubuntu so I'm building from source and my practice is to add include directories and libs directly into the cmake file. It works but it could be better. I have an example using make and emcc compiler that should work with clang compiler as well [https://github.com/jnmaloney/YourWebFilament](https://github.com/jnmaloney/YourWebFilament).

**Reply by bigianb** [reply]
Thank you, that's helpful.
